articles/azure-monitor/app/javascript-troubleshoot.md

```diff
@@ -0,0 +1,177 @@
+>>
+>> Here is an example of how this may be configured within [Chrome to allow or block access to websites](https://support.google.com/chrome/a/answer/7532419?hl=en)
+> ---
+
+> ##### Corporate Exceptions for CDN endpoints
+>
+>> This is similar to [adding exceptions for end-users](add-exceptions-for-cdn-endpoints), but you will need to work with your end-users IT department to have them configure the Application Insights CDN endpoints to be downloaded by including (or removing) them in any black or white list.
+>>
+>> __Special Note:__ If the your corporate user is using a [private cloud](https://azure.microsoft.com/overview/what-is-a-private-cloud/) and they cannot enable any form of exception to provide their internal users with public access for the CDN endpoints then you will need to [use the NPM packages](#use-npm) or [host the Application Insights SDK on your own CDN](#host-sdk).
+> ---
+
+> ##### Use NPM
+> 
+>> Rather than using the snippet and public CDN endpoints you could can use the NPM packages to include the SDK as part of your own JavaScript files. With this approach the SDK will become just another package within your own scripts. Note: It is recommended that when using this approach, you should also use some form of [javascript bundler](https://www.bing.com/search?q=javascript+bundler) to assist with code splitting and minification.
+>>
+>> As with the snippet, it is also possible that your own scripts (with or without using the SDK NPM packages) could be affected by the same blocking issues listed here, so depending on your application; users and framework you may want to consider implementing something similar to logic in the snippet to detect and report these issues.
+>>
+>> Note: You will need to use this approach (or [Host the SDK on you own CDN](#host-sdk)) if your users are using a [private cloud](https://azure.microsoft.com/overview/what-is-a-private-cloud/) as they most likely will not have access to the public internet.
+> ---
+
+> ##### Host SDK
+> 
+>> Rather than your end-users downloading the Application Insights SDK from the public CDN you could host the Application Insights SDK from your own CDN endpoint. If using this approach it is recommended that you use a specific version (ai.2.#.#.min.js) so that it's easier to identify which version you are using, it is also recommended that update it on a regular basis to the current version (ai.2.min.js) so you can leverage any bug fixes and new features that become available.
+>>
+>> Note: You will need to use this approach (or [Use NPM packages](#use-npm)) if your users are using a [private cloud](https://azure.microsoft.com/overview/what-is-a-private-cloud/) as they most likely will not have access to the public internet.
+> ---
+
+> ##### CDN Blacklisted
```

Comment:
This is actually a "diagnose" step mixed in with the troubleshooting steps. I suggest you break it out and add it to the article above under a #####diagnose header.

Reply:
There is no distinction between diagnosis and solution (i.e. collectively troubleshooting), they are grouped more deliberately as possible solutions as the developers can choose which option they want to try.

So generally I've tried to phrased then as 
- if you have this (or can confirm this) then 
- this is what you should (or can) do

Where the "Known Issues" are more of
- description of the problem
- collection of possible solutions for this case

Where this one originally item only had create-ticket link, I'm updating to include a new possible solution.

Reply:
This one while it could be seen as a issue, is referenced from multiple issues. So it contains what I'm now calling the "toubleshooting steps" as suggested.